Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,771 @@
+                    }
+
+                    if (lightType === 'spot') {
+                        fragmentLightingBlock += '    }\n';
+                    }
+                }
+                fragmentLightingBlock += '  }\n';
+            }
+        }
+
+        if (!hasAmbientLights) {
+            // Add an ambient light if we don't have one
+            fragmentLightingBlock += '  ambientLight += vec3(0.1, 0.1, 0.1);\n';
+        }
+
+        if (!hasNonAmbientLights) {
+            fragmentLightingBlock += '  vec3 l = normalize(czm_sunDirectionEC);\n';
+            fragmentLightingBlock += '  diffuseLight += vec3(1.0, 1.0, 1.0) * max(dot(normal,l), 0.);\n';
+
+            if (hasSpecular) {
+                if (lightingModel === 'BLINN') {
+                    fragmentLightingBlock += '  vec3 h = normalize(l + viewDir);\n';
+                    fragmentLightingBlock += '  float specularIntensity = max(0., pow(max(dot(normal, h), 0.), u_shininess));\n';
+                }
+                else { // PHONG
+                    fragmentLightingBlock += '  vec3 reflectDir = reflect(-l, normal);\n';
+                    fragmentLightingBlock += '  float specularIntensity = max(0., pow(max(dot(reflectDir, viewDir), 0.), u_shininess));\n';
+                }
+
+                fragmentLightingBlock += '  specularLight += vec3(1.0, 1.0, 1.0) * specularIntensity;\n';
+            }
+        }
+
+        vertexShader += 'void main(void) {\n';
+        vertexShader += vertexShaderMain;
+        vertexShader += '}\n';
+
+        fragmentShader += 'void main(void) {\n';
+        var colorCreationBlock = '  vec3 color = vec3(0.0, 0.0, 0.0);\n';
+        if (hasNormals) {
+            fragmentShader += '  vec3 normal = normalize(v_normal);\n';
+            if (isDoubleSided) {
+                fragmentShader += '  if (gl_FrontFacing == false)\n';
+                fragmentShader += '  {\n';
+                fragmentShader += '    normal = -normal;\n';
+                fragmentShader += '  }\n';
+            }
+        }
+
+        var finalColorComputation;
+        if (lightingModel !== 'CONSTANT') {
+            if (defined(techniqueParameters.diffuse)) {
+                if (techniqueParameters.diffuse.type === WebGLConstants.SAMPLER_2D) {
+                    fragmentShader += '  vec4 diffuse = texture2D(u_diffuse, ' + v_texcoord + ');\n';
+                }
+                else {
+                    fragmentShader += '  vec4 diffuse = u_diffuse;\n';
+                }
+                fragmentShader += '  vec3 diffuseLight = vec3(0.0, 0.0, 0.0);\n';
+                colorCreationBlock += '  color += diffuse.rgb * diffuseLight;\n';
+            }
+
+            if (hasSpecular) {
+                if (techniqueParameters.specular.type === WebGLConstants.SAMPLER_2D) {
+                    fragmentShader += '  vec3 specular = texture2D(u_specular, ' + v_texcoord + ');\n';
```

Comment:
I had a problem here in the obj converter, I think it should be: 
`vec3 specular = texture2D(u_specular, ' + v_texcoord + ').rgb`